### ISSUE TITLE
org.openrewrite.java.cleanup.RenamePrivateFieldsToCamelCase does not recognize @Data lombok annotation

### ISSUE THREAD

**Issue by blipper** [openrewrite/rewrite-static-analysis#8]
org.openrewrite.java.cleanup.RenamePrivateFieldsToCamelCase only checks if there is a private field.  It does not recognize @Data or any other lombok annotation so renaming the field will introducing a breaking change.  To preserve compatibility it should ignore private fields if the class is annotated with @Data/@Getter/@Setter

**Comment by timtebeek** [comment]
@gwydionmv added a very similar change in 
- https://github.com/openrewrite/rewrite-static-analysis/pull/244

That could be replicated here to make `RenamePrivateFieldsToCamelCase` skip classes or fields with Lombok annotations.

**Comment by timtebeek** [comment]
As part of that change we'd then likely want to do the same for ExplicitInitializationVisitor
https://github.com/openrewrite/rewrite-static-analysis/blob/a5dc2f7049f7ef4554180c630af75985d388ac4b/src/main/java/org/openrewrite/staticanalysis/ExplicitInitializationVisitor.java#L59-L62

**Comment by gwydionmv** [comment]
@timtebeek I can take care of this too :D

**Comment by blipper** [comment]
Ultimately what think is needed for all of these annotation processors is a mechanism where rewrite has access to the paired byte code generated at compilation time.  AST analysis will never be enough with annotation processors.

**Comment by blipper** [comment]
Or a at a minimum if you want to keep pure AST analysis you need to pair with the delomboked source

**Comment by timtebeek** [comment]
Thanks for helping think this one through! The reason I'd want to skip lombok annotated classes and fields is that this recipe also changes snake case to camel case, as seen in this test case
https://github.com/openrewrite/rewrite-static-analysis/blob/541198c276fc0e5b45c6a9a9d805e23779be2267/src/test/java/org/openrewrite/staticanalysis/RenamePrivateFieldsToCamelCaseTest.java#L45-L50

I'm not a 100% sure what getters lombok would generate before and after the rename, but I think that'd be `getD_TYPE_CONNECT()`, so best skipped.

**Comment by gwydionmv** [comment]
yes, yes, you're right, that's what Lombok generates for that field, `.getD_TYPE_CONNECT()`. And that's what my point was, if that variable name is wrong, then it's ok that the recipe _breaks_ the code, isn't it? It won't be as automatic as other recipes, but I don't find it a bad thing that after running the recipe the dev has to replace the old getter with `getDTypeConnect()`

**Comment by timtebeek** [comment]
I understand where you're coming from in that if you want to thoroughly clean up the code in a single repository, that it's then perhaps acceptable to have to put some effort as well. However, we strongly belief in keeping to our [do no harm](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices#do-no-harm) convention, where we only make a code change when we believe it safe to do so.

Consider for example that you're not looking at a single project but dozens or even hundreds; and then not running a single recipe, but also dozens or hundreds. In those cases you'd much rather have each of those recipes _not_ make a change when there's potential to break code, as there would be a lot of code to check afterwards. Or consider the case where you run the recipe on a library, that may then subtly change the exposed getter, which you might only notice in consumers.

Given the above I think it's best to make all the code changes we know are safe, and back off from making code changes when there might be unhandled side effects such as in the case of Lombok annotated fields. That's also why we for instance don't run our "[Convert @lombok.Value class to Record](https://docs.openrewrite.org/recipes/java/migrate/lombok/lombokvaluetorecord)" recipe by default. As much as we think it might improve code, there's also potential for negative effects in downstream consumers of libraries, so we leave it up to the user to consider running that by itself as desired.

**Comment by blipper** [comment]
Working at Amazon and previously worked at Google which both operate at scale you need to minimize false positives so +1.

That being said with Lombok the private fields are similar to records is that they become a public API/source of truth so arguably they aren't private at all anymore.

**Comment by timtebeek** [comment]
We now support Lombok, so this should no longer be an issue; thanks all!
- https://github.com/openrewrite/rewrite/issues/1297
